ExtractorUtils/BaseExtractor.cs

```diff
@@ -84,17 +97,34 @@ public virtual async Task Start(CancellationToken token)
             try
             {
                 await Start().ConfigureAwait(false);
+                if (Run != null)
+                {
+                    Run.Start();
+                }
                 await Scheduler.WaitForAll().ConfigureAwait(false);
             }
+            catch (Exception ex)
+            {
+                if (token.IsCancellationRequested) return;
+                if (Run != null)
+                {
+                    await Run.Report(ExtPipeRunStatus.failure, true,
+                        $"Error: {ex.Message}\n{ex.StackTrace}", token).ConfigureAwait(false);
+                }
+                throw;
+            }
             finally
             {
                 await OnStop().ConfigureAwait(false);
+                await Run.DisposeAsync().ConfigureAwait(false);
```

Comment:
Should there be a null-check for Run here as well?

Reply:
Yes. Good catch.